source/globalCommands.py

```diff
@@ -4936,6 +4944,240 @@ def _enableScreenCurtain(doEnable: bool = True):
 					return
 				_enableScreenCurtain()
 
+	# Instance of NVDA magnifier
+
+	_nvdaMagnifier: FullScreenMagnifier | DockedMagnifier | LensMagnifier | None = None
+	_colorFilter: ColorFilter = ColorFilter.NORMAL
+	_magnifierType: MagnifierType = MagnifierType.FULLSCREEN
+	_fullscreenMode: FullScreenMode = FullScreenMode.CENTER
+	_zoomLevel: float = 2.0
+
+	@script(
+		description=_(
+			# Translators: Describes a command.
+			"Toggles the magnifier on and off."
+		),
+		category=SCRCAT_VISION,
+		gestures=["kb:windows+numLock+numpadPlus", "kb:windows+numpadPlus", "kb:windows+="],
+	)
+	def script_startMagnifier(self, gesture):
+		nvdaMagnifier = self._nvdaMagnifier
+		if nvdaMagnifier and nvdaMagnifier.isActive:
+			nvdaMagnifier._zoom(True)
+			self._zoomLevel = nvdaMagnifier.zoomLevel
+		elif nvdaMagnifier is None:
+			log.info(f"""
+			Starting NVDA magnifier with
+			Magnifier Type: {self._magnifierType.name.lower()}
+			Zoom Level: {self._zoomLevel}
+			Color Filter: {self._colorFilter.name.lower()}
+			Fullscreen Mode: {self._fullscreenMode.name.lower()}
+			""")
+			if self._magnifierType == MagnifierType.FULLSCREEN:
+				self._nvdaMagnifier = FullScreenMagnifier(
+					self._zoomLevel, self._colorFilter, self._fullscreenMode
+				)
+			elif self._magnifierType == MagnifierType.DOCKED:
+				self._nvdaMagnifier = DockedMagnifier(self._zoomLevel, self._colorFilter)
+			elif self._magnifierType == MagnifierType.LENS:
+				self._nvdaMagnifier = LensMagnifier(self._zoomLevel, self._colorFilter)
+			ui.message(
+				_(
+					# Translators: Message announced when starting the NVDA magnifier with {type} being the magnifier type
+					"Starting NVDA {type} magnifier"
+				).format(type=self._magnifierType.name.lower())
+			)
+		else:
+			log.info("error starting")
+			ui.message(
+				_(
+					# Translators: Message announced when failing to start the NVDA magnifier
+					"Error starting NVDA magnifier"
+				)
+			)
+
+	@script(
+		description=_(
+			# Translators: Describes a command.
+			"decreases the magnification level of the screen magnifier."
+		),
+		category=SCRCAT_VISION,
+		gestures=["kb:windows+numLock+numpadMinus", "kb:windows+numpadMinus", "kb:windows+$"],
+	)
+	def script_zoomOut(self, gesture):
+		nvdaMagnifier = self._nvdaMagnifier
+		if nvdaMagnifier and nvdaMagnifier.isActive:
+			nvdaMagnifier._zoom(False)
+			self._zoomLevel = nvdaMagnifier.zoomLevel
+		else:
+			return
+
+	@script(
+		description=_(
+			# Translators: Describes a command.
+			"Stop the magnifier"
+		),
+		category=SCRCAT_VISION,
+		gesture="kb:windows+escape",
+	)
+	def script_StopMagnifier(self, gesture):
```

Comment:
lets remove this and just use a single `script_toggleMagnifier` rather than `script_startMagnifier` or `script_StopMagnifier`. That way a single gesture turns the magnifier on or off. This should also be a setting, so magnifier starts by default with NVDA starts if desired.